mitogen/master.py

```diff
@@ -706,21 +706,20 @@ def on_shutdown(self, broker):
     def _first_stage():
         import os,sys,zlib
         R,W=os.pipe()
-        R2,W2=os.pipe()
+        r,w=os.pipe()
         if os.fork():
             os.dup2(0,100)
             os.dup2(R,0)
-            os.dup2(R2,101)
-            for f in R,R2,W,W2: os.close(f)
-            os.environ['ARGV0'] = sys.executable
-            os.execv(sys.executable,['mitogen:CONTEXT_NAME'])
-        else:
-            os.write(1, 'EC0\n')
-            C = zlib.decompress(sys.stdin.read(input()))
-            os.fdopen(W,'w',0).write(C)
-            os.fdopen(W2,'w',0).write('%s\n%s' % (len(C),C))
-            os.write(1, 'EC1\n')
-            sys.exit(0)
+            os.dup2(r,101)
+            for f in R,r,W,w:os.close(f)
```

Comment:
`str(map(os.close,(R,r,W,w)))` has the same character count, but would eliminate the `for` block. This would allow adding a step to `get_boot_command()` that munges the entire `if` block into a single line, with `;` delimiters, saving some `\n` bytes.

The `str()` wrapper would be necessary to 'pump' the `map` builtin on Python 3.